docs/DeveloperGuide.md

```diff
@@ -2,12 +2,27 @@
 
```

Comment:
Perhaps the Developer Guide title could include the application's name (e.g. Developer Guide for AlloNUS) so that the reader will know that he/she is reading the right document.